pkg/lib/ssh-setup-default-key.sh

```diff
@@ -0,0 +1,21 @@
+#! /bin/sh
+
+set -euf
+
+# Setup the default SSH key.  This consists of creating it if
+# necessary, or changing its passphrase to match the login password.
+# Then we dump the corresponding public key to stdout.
+
+file=$1
+cur_passphrase=$2
+login_password=$3
+
+if ! test -e "$file"; then
+    dir=`dirname "$file"`
+    test -e "$dir" || ( mkdir "$dir" && chmod 700 "$dir" )
+    ssh-keygen -q -t rsa -f "$file" -N "" >/dev/null
```

Comment:
`-t rsa` is the default, and IMHO we shouldn't override it in this script. The admin may have configured ECC or something else to be used by default.

Let's not generate a passwordless key by default, but at least use the login password by default (or a custom passphrase if specified). We want to establish best practices here, and passwordless ssh keys are definitively not that.

Reply:
The type is connected to the file name though.  When creating a new key, it will always be `~/.ssh/id_rsa` and I feel it would be wrong for it to have a different type than RSA.  I have made it so that now the name and type are specified in the same place (machine-dialogs.js).

Reply:
> Let's not generate a passwordless key by default, but at least use the login password by default (or a custom passphrase if specified). We want to establish best practices here, and passwordless ssh keys are definitively not that.

Done.